### PR TITLE
[AC-5898] Eliminate naive datetime

### PR DESCRIPTION
**Pull request by jonkiparsky**
Converting the `datetime` to a `date` strips tz info, causing a naive datetime to be created incorrectly.  
Came across this while working on a test for AC-5836, so fixed it. 

To test: 
check out impact-api, run tests. Note timezone warning
point impact-api at this branch, build, and run tests again. Note, no timezone warning

### CODE REVIEW

**Comment by jonkiparsky** [comment]
@yotammanor It seems very odd that this change would have caused these issues. Any theories about what's happening here?

**Comment by jonkiparsky** [comment]
Digging in, it looks like the ExpertMutationForm is coming back invalid after this change. 
Looking further.

**Comment by jonkiparsky** [comment]
Verified that this is a testing issue, functionality works on development with this change

**Comment by jonkiparsky** [comment]
@yotammanor Found a way around the problem.

File: accelerator/tests/factories/factory_utils.py [hunk 1]
```diff
@@ -28,5 +28,5 @@ def expert_data(user, password="password"):
         "public_website_consent": True,
         "username": user.username,
         "password": password,
-        "date_joined": user.date_joined,
+        "date_joined": user.date_joined.date(),
```
Comment (yotammanor) [inline accelerator/tests/factories/factory_utils.py]: 😱

**Pull request merged**